File: mobile/lib/shared/models/membership.dart
class Membership {
  const Membership({
    required this.id,
    required this.memberId,
    this.planId,
    this.planName,
    required this.startDate,
    required this.expiryDate,
    required this.price,
    required this.status,
  });

  factory Membership.fromJson(Map<String, dynamic> json) => Membership(
        id: json['id'] as int,
        memberId: json['memberId'] as int,
        planId: json['planId'] as int?,
        planName: json['planName'] as String?,
        startDate: json['startDate'] as String? ?? '',
        expiryDate: json['expiryDate'] as String? ?? '',
        price: (json['price'] as num?)?.toInt() ?? 0,
        status: json['status'] as String? ?? 'Active',
      );

  final int id;
  final int memberId;
  final int? planId;
  final String? planName;
  final String startDate;
  final String expiryDate;
  final int price;
  final String status;
}

File: backend/app/Models/Membership.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToTenant;
use Database\Factories\MembershipFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

#[Fillable(['member_id', 'plan_id', 'start_date', 'expiry_date', 'price', 'status'])]
class Membership extends Model
{
    /** @use HasFactory<MembershipFactory> */
    use BelongsToTenant, HasFactory;

    protected function casts(): array
    {
        return [
            'start_date' => 'date',
            'expiry_date' => 'date',
        ];
    }

    public function member(): BelongsTo
    {
        return $this->belongsTo(Member::class);
    }

    public function plan(): BelongsTo
    {
        return $this->belongsTo(MembershipPlan::class, 'plan_id');
    }
}
